serverless/azure-functions/node/http-trigger-app/open-telemetry/tracing.js

```diff
@@ -0,0 +1,182 @@
+    // (Trace flags set to 00), this example uses the web headers directly.
+    const headers = request.headers;
+    const incomingContext = opentelemetry.propagation.extract(opentelemetry.ROOT_CONTEXT, headers);
+
+    // Name should be Azure Function Application name and Function name.
+    const appAndFunctionName = `${functionAppName}/${executionContext.functionName}`;
+
+    // NOTE: There is a lot more HTTP information that can be added than plumbed in this example for
+    // request and response attributes defined in the http semantic conventions. FaaS conventions are currently
+    // experimental and subject to change. For FaaS and Http Semantic contentions, see the following specifications:
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/faas.md
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/faas.md
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md
+    const span = tracer.startSpan(
+      executionContext.functionName,
+      {
+        kind: opentelemetry.SpanKind.SERVER,
+        attributes: {
+          // FAAS attributes are not currently needed/required by New Relic but may
+          // drive experiences in the future and are part of the semantic conventions.
+          [SemanticResourceAttributes.FAAS_NAME]: appAndFunctionName,
+          [SemanticAttributes.FAAS_TRIGGER]: FaasTriggerValues.HTTP,
+          [SemanticAttributes.FAAS_EXECUTION]: executionContext.invocationId,
+          [SemanticAttributes.FAAS_COLDSTART]: _isColdStart(),
+          [SemanticAttributes.HTTP_SCHEME]: 'http',
+          [SemanticAttributes.HTTP_METHOD]: request.method
+        }
+      },
+      incomingContext
+    );
+
+    let _error = null;
+    try {
+      const spanContext = opentelemetry.trace.setSpan(opentelemetry.context.active(), span);
+      await opentelemetry.context.with(spanContext, async () => {
+        return originalFunction.apply(this, arguments);
+      });
+    } catch (error) {
+      _error = error;
+
+      // Re-throwing original error.
+      throw error;
+    } finally {
+      await _endInvocation(span, _error, context.res.status);
+    }
+  }
+}
+
+async function _endInvocation(span, error, status) {
+  if (error) {
+    span.recordException(error);
+
+    // Defaulting to 500 when the function throws.
+    span.setAttributes({
+      [SemanticAttributes.HTTP_STATUS_CODE]: 500
+    });
+
+    span.setStatus({
+      code: opentelemetry.SpanStatusCode.ERROR,
+      message: error.message
+    });
+  } else {
+    const statusCode = status && parseInt(status); // Should have some error safety here.
+
+    span.setAttributes({
+      [SemanticAttributes.HTTP_STATUS_CODE]: statusCode
+    });
+
+    // Your threshold considerations may vary.
+    if (status >= 400) {
```

Comment:
Use status >= 500. This aligns with otel's http semantic conventions https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#status

Reply:
Can do. I saw that was OTel standard but New Relic's approach in existing agents is 400+ so I went with the consistent experience. I guess we can reconcile all this at some point in the future.